acme/acme/challenges.py

```diff
@@ -318,89 +319,36 @@ def validation(self, account_key):
         return self.key_authorization(account_key)
 
 
-@Challenge.register  # pylint: disable=too-many-ancestors
-class DVSNI(_TokenDVChallenge):
-    """ACME "dvsni" challenge.
-
-    :ivar bytes token: Random data, **not** base64-encoded.
-
-    """
-    typ = "dvsni"
-
-    PORT = 443
-    """Port to perform DVSNI challenge."""
-
-    def gen_response(self, account_key, alg=jose.RS256, **kwargs):
-        """Generate response.
-
-        :param .JWK account_key: Private account key.
-        :rtype: .DVSNIResponse
-
-        """
-        return DVSNIResponse(validation=jose.JWS.sign(
-            payload=self.json_dumps(sort_keys=True).encode('utf-8'),
-            key=account_key, alg=alg, **kwargs))
-
-
 @ChallengeResponse.register
-class DVSNIResponse(ChallengeResponse):
-    """ACME "dvsni" challenge response.
-
-    :param bytes s: Random data, **not** base64-encoded.
-
-    """
-    typ = "dvsni"
+class TLSSNI01Response(KeyAuthorizationChallengeResponse):
+    """ACME tls-sni-01 challenge response."""
+    typ = "tls-sni-01"
 
     DOMAIN_SUFFIX = b".acme.invalid"
     """Domain name suffix."""
 
-    PORT = DVSNI.PORT
-    """Port to perform DVSNI challenge."""
-
-    validation = jose.Field("validation", decoder=jose.JWS.from_json)
+    PORT = 443
 
     @property
-    def z(self):  # pylint: disable=invalid-name
-        """The ``z``  parameter.
-
-        :rtype: bytes
-
-        """
-        # Instance of 'Field' has no 'signature' member
-        # pylint: disable=no-member
-        return hashlib.sha256(self.validation.signature.encode(
-            "signature").encode("utf-8")).hexdigest().encode()
+    def z(self):
+        """``z`` value used for verification."""
+        return hashlib.sha256(
+            self.key_authorization.encode("utf-8")).hexdigest().encode()
```

Comment:
Interestingly, Python docs doesn't guarantee it to be lower case, so I've added `.lower()`.

https://docs.python.org/2/library/hashlib.html#hashlib.hash.hexdigest